### PR TITLE
Added `onClear` prop to `EuiFormControlLayout`

### PR DESCRIPTION
**Pull request by cchaos**
There's been talk about replacing the browser-specific clear functions in inputs like search types with our own implementation, so I just added an option to the `EuiFormControlLayout` to allow for this function and if it exists, it will display the proper button.

We don't really want to add any number of icons to this layout control because we can't then properly account for it with the padding on the input. We should still restrict to just one icon and then these extra functions will display if needed.

Also, I removed the toggling of the caret and after talking with Dave a bit, decided to just leave the default hover cursor (text) when hovering the caret.

### CODE REVIEW

File: src/components/combo_box/combo_box.js [hunk 1]
```diff
@@ -40,13 +40,11 @@ export class EuiComboBox extends Component {
     renderOption: PropTypes.func,
     isInvalid: PropTypes.bool,
     rowHeight: PropTypes.number,
-    clearable: PropTypes.bool,
```
Comment (nreese) [inline src/components/combo_box/combo_box.js:43]: This prop is still needed. There needs to be a way to tell a ComboBox that the clear icon should not be displayed. For example, Visual Builder uses react-select and sets `clearable` property to ensure certain values are never unset.

File: src/components/combo_box/combo_box.js [hunk 2]
```diff
@@ -567,10 +564,6 @@ export class EuiComboBox extends Component {
           autoSizeInputRef={this.autoSizeInputRef}
           inputRef={this.searchInputRef}
           updatePosition={this.updateListPosition}
-          isListOpen={isListOpen}
-          clearable={clearable}
-          onOpen={this.openList}
-          onClose={this.closeList}
```
Comment (nreese) [inline src/components/combo_box/combo_box.js]: What about being able to close the list by clicking the icon? I think this is a nice feature to have.

File: src/components/combo_box/combo_box.js (hunk 1 above)
Comment (cchaos) [reply src/components/combo_box/combo_box.js:43]: Ah sorry, I'll put that back in.

File: src/components/combo_box/combo_box.js (hunk 2 above)
Comment (cchaos) [reply src/components/combo_box/combo_box.js]: Hm, yeah ok, I'll figure out a way to signal that an icon is a button in the form layout.

**Comment by cchaos** [comment]
Ok I added `isClearable` and onClose/onOpen back in passing through an `onIconClick` prop on the EuiFormControlLayout that will then wrap the icon in a button.

**Pull request merged**